shell.nix

```
@@ -4,4 +4,5 @@ with pkgs;
 
 mkShell {
   inputsFrom = lib.attrValues (import ./. { inherit pkgs; });
+  HPOS_STATE_PATH = "blah";
```

:smirk_cat: 

```suggestion
```

---

This should be just not set, right? In HPOS this will be handled, and hp-admin-authorize-server can handle it gracefully in the nix-shell where `HPOS_STATE_PATH` is not set.

---

In a shell.nix development context, you may want to set it. Your mileage may vary.

In hpos, yes we would set it in a profile in holo-nixpkgs

---

It's OK to set it, but it shouldn't be to set to a meaningless value like `blah` (which is worse than setting it to no value at all). If you want to provide a pre-generated `hpos-state.json` so that developer can immediately drop into `nix-shell` and start developing this, commit `hpos-state.json` into this repo and change this line to something like:

```suggestion
  HPOS_STATE_PATH = "${./server/hpos-state.json}";
```

Alternative is to not set this at all.

---

Right sorry. I meant to return to this and understand what it should be set to in a shell.nix context. Sorry for forgetting that. Thanks for that example.

---

I will check again, but I believe the software would not build without this value set.

---

In the current implementation I provide test file in `https://github.com/Holo-Host/hp-admin-crypto/tree/2019-11-auth-server/server/resources/test`. So that would be:
```
HPOS_STATE_PATH = "${./server/resources/test/hpos-state.json}";
```

---

Sam updated me that he's OK with this change.

---

Ok, pushing this

---

oh, you did :-)

---

Yego is going to just push some commits and merge do that we can all move in with our lives 😀